이벤트 추가 시 사진 추가 기능

Yo0oN opened the pull request:
- [x]  이벤트 추가 시 이미지 파일 존재하면 AWS S3에 저장하는 기능 추가

- [x]  게시글 삭제 시 파일 삭제 기능은 이후 추가 예정

Yo0oN committed:
[#101] 이벤트 추가 시 사진 추가 기능

이미지 추가 시 AWS S3에 저장하는 기능 추가

Jane096 commented on src/main/java/com/festa/common/commonService/ImageUploader.java:
```diff
@@ -0,0 +1,15 @@
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+
+@Service
+public interface ImageUploader {
```
여기랑 S3ImageUploader에 둘 다 `@Service` 가 선언되어있는데 혹시 선언하신 이유가 있으신가요?

Yo0oN replied:
이후 S3가 아닌 다른 이미지 서버를 만들것을 생각해서 Interface를 만들어두었습니다.
이때 `ImageUploader`에 `@Service`를 사용하고, 구현받은 `S3ImageUploader`에서 `@Component`를 사용해야 나중에 `ImageUploader` 타입으로 빈을 주입받을 때 `ImageUploader`를 구현한 타입을 주입해주기 때문에 선언하였습니다!

Jane096 commented on src/main/java/com/festa/controller/EventController.java:
```diff
@@ -15,15 +16,8 @@
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
```
이거 아마 인텔리제이에서 자동으로 `*` 으로 import 되는 경우가 있는 것 같은데 single import로 하는게 좋아보입니다!

Yo0oN replied:
넵 이부분은 수정하였습니다!

Jane096 commented on src/main/java/com/festa/controller/EventController.java:
```diff
@@ -137,8 +137,13 @@
-    public void modifyEventsInfo(@RequestBody EventDTO eventDTO, @CurrentLoginUserNo long userNo) {
-        eventService.modifyEventsInfo(eventDTO, userNo);
+    public void modifyEventsInfo(@RequestBody EventDTO eventDTO, @RequestPart MultipartFile multipartFile, @CurrentLoginUserNo long userNo) {
+        String fileName = null;
+        if (!multipartFile.isEmpty()) {
+            fileName = s3ImageUploader.uploadImage(multipartFile);
+        }
```
`fileName` 을 null로 먼저 초기화 해주셨는데 이렇게 하신 이유가 있으신가요?

Yo0oN replied:
이부분은 파일이 있는경우에만 `uploadImage`를 수행하여 fileName이 초기화되기 때문에, 조건문 밖에서 먼저 초기화를 해주기 위해 null로 설정해주었습니다.
하지만 이럴경우 아래의 `modifyEventsInfo`의 매개변수에 null이 들어가게 되는 일이 발생하기 때문에 좋은 방식은 아닌것같아 수정해보려고 생각중입니다..!

Yo0oN committed:
[#101] 사진 추가 기능 수정

application.properties에 암호화된 Key 추가
EventController에서 S3ImageUploader타입이 아닌 ImageUploader타입으로 받아오도록 수정

Yo0oN committed:
[#101] AWS 지역 추가

f-lab-dev commented on src/main/java/com/festa/common/S3ImageUploader.java:
```diff
@@ -0,0 +1,123 @@
+
+    private AmazonS3 amazonS3;
+
+    @Value("aws.s3.bucket")
```
이런 값들도 constructor injection이 가능합니다~

f-lab-dev commented on src/main/java/com/festa/common/S3ImageUploader.java:
```diff
@@ -0,0 +1,123 @@
+        AWSCredentials awsCredentials = new BasicAWSCredentials(accessKey, secretKey);
+
+        amazonS3 = AmazonS3ClientBuilder.standard()
+                .withRegion("ap-northeast-2")
```
리전은 영원히 이 값으로 고정되는걸까요?

Yo0oN replied:
영원히 고정은 아니라고 생각합니다..!이부분은 설정파일에서 가져올수있도록 수정하겠습니다!

Yo0oN committed:
[#101]region값 수정

이후 변경되어도 수정하기 편하도록 application.properties에 설정

Yo0oN committed:
Merge branch 'develop' into feature/#101
